src/nspawn/nspawn.c

```diff
@@ -3551,15 +3551,15 @@ static int inner_child(
         envp[n_env++] = strjoina("container=", arg_container_service_name);
 
         /* Propagate $TERM & Co. unless we are invoked in pipe mode and stdin/stdout/stderr don't refer to a TTY */
-        if (arg_console_mode != CONSOLE_PIPE || on_tty()) {
+        if (on_tty())
```

Comment:
hmm, shouldn't this better be:

`if (arg_console_mode != CONSOLE_PIPE && on_tty())` 

i.e. propagate the three if we both are on a tty and and we are invoking things on a tty, otherwise set things to dumb.

might even go further and use !terminal_is_dumb() instead of on_tty() here()...